src/transformers/models/gptsan_japanese/modeling_gptsan_japanese.py

```diff
@@ -0,0 +1,1246 @@
+        "武田信玄は、戦国の頃より「智勇兼備」した英雄として織田信長に比されてきた戦国武将であり、..."
+        ```"""
+        SEP_TOKEN = self.config.pad_token_id
+        NOT_TOKEN = self.config.unk_token_id
+        use_cache = use_cache or self.config.use_cache
+        return_dict = return_dict or self.config.return_dict
+        model_return_dict = True
+        model_output_hidden_states = True
+
+        outputs = self.model(input_ids,
+                            num_precontext,
+                            spout,
+                            past_key_values,
+                            head_mask,
+                            use_cache,
+                            inputs_embeds,
+                            decoder_inputs_embeds,
+                            output_attentions,
+                            model_output_hidden_states,
+                            model_return_dict,
+                            output_router_logits)
+
+        hidden_states = self.lm_head(outputs[0])
+        hidden_states = self.act(hidden_states)
+
+        lm_logits = torch.einsum("bsc,vc->bsv", hidden_states, self.model.embed_tokens.weight)
+        if lm_logits.shape[-1] == self.final_logits_bias.shape[-1]:
+            lm_logits = lm_logits + self.final_logits_bias
+
+        loss = None
+        z_loss = None
+        router_probs = None
+        aux_loss = None
+        if labels is not None:
+            loss_fct = nn.CrossEntropyLoss(ignore_index=-100)
+
+            if output_router_logits:
+                # Compute the router loss (z_loss + auxiliary loss) for each router in the encoder and decoder
+                router_logits, expert_indexes = self._unpack_router_logits(
+                    encoder_outputs.router_probs
+                )
+                z_loss = router_z_loss_func(encoder_router_logits)
+                router_probs = nn.Softmax(dim=-1)(encoder_router_logits)
+                aux_loss = load_balancing_loss_func(encoder_router_probs, encoder_expert_indexes)
+
+            loss = loss_fct(lm_logits.view(-1, lm_logits.size(-1)), labels.view(-1))
+
+        # Ignore the two tokens when generating
+        lm_logits[:, :, SEP_TOKEN] = -1e10
+        lm_logits[:, :, NOT_TOKEN] = -1e10
+
+        if not return_dict:
+            return tuple(
+                v
+                for v in [
+                    loss,
+                    lm_logits,
+                    #z_loss,
+                    #aux_loss,
+                    outputs.past_key_values,
+                    #outputs.last_hidden_state,
+                    outputs.hidden_states,
+                    outputs.attentions,
+                    #outputs.router_probs,
+                ]
+                if v is not None
+            )
+
+        return CausalLMOutputWithPast(
+            loss=loss,
+            logits=lm_logits,
+            past_key_values=outputs.past_key_values,
+            hidden_states=outputs.hidden_states,
+            attentions=outputs.attentions,
+        )
```

Comment:
To answer your question here, you can create a new type `MOECausalLMOutputWithPast` and simply add the variable that you need (since it is not an encoder decoder) 😉